tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -17,65 +17,33 @@ import org.scalajs.core.ir.Trees.FieldDef
 import org.scalajs.core.tools.linker._
 
 private[emitter] object KnowledgeGuardian {
-  private[KnowledgeGuardian] trait Unregisterable {
+  trait Unregisterable {
     def unregister(invalidatable: Invalidatable): Unit
   }
 
   trait Invalidatable {
-    private[KnowledgeGuardian] def registeredTo(
-        unregisterable: Unregisterable): Unit
+    def registeredTo(unregisterable: Unregisterable): Unit
```

Comment:
If it were truly public, I would agree. But here it's only "public" up to `emitter`, so it's not really a big deal. IMO it's better to keep them in the companion to avoid the extra outer pointers.